src/mlpack/methods/naive_bayes/naive_bayes_classifier.hpp

```diff
@@ -123,7 +123,7 @@ class NaiveBayesClassifier
    * @param data List of data points.
    * @param results Vector that class predictions will be placed into.
    */
-  void Classify(const MatType& data, arma::Row<size_t>& results);
+  void Classify(const MatType& data, arma::Row<size_t>& results, MatType& results_proba);
```

Comment:
This will have to be of type `arma::mat` not `MatType`---if the user sets `MatType` to, i.e. `arma::umat`, we can't return posteriors as a `umat`, we need double-precision.